main.tf

```diff
@@ -0,0 +1,39 @@
+module "label" {
+  source     = "git::https://github.com/cloudposse/tf_label.git?ref=tags/0.2.0"
+  namespace  = "${var.namespace}"
+  stage      = "${var.stage}"
+  name       = "${var.name}"
+  attributes = "${var.attributes}"
+  delimiter  = "${var.delimiter}"
+  tags       = "${var.tags}"
+}
+
+resource "aws_key_pair" "non_empty" {
+  count      = "${signum(length(var.ssh_public_key_path)) == 1 ? 1 : 0}"
+  key_name   = "${module.label.id}"
+  public_key = "${file("${var.ssh_public_key_path}")}"
+}
+
+resource "tls_private_key" "default" {
```

Comment:
nice! I did not know about this resource type.

Reply:
I did not know it too before today